WordPress/Classes/Models/Blog+Lookup.swift

```diff
@@ -69,6 +69,42 @@ public extension Blog {
         try? BlogQuery().xmlrpc(matching: xmlrpc).selfHostedBlogUsername(username).blog(in: context)
     }
 
+    /// Searches for a `Blog` object for this account with the given XML-RPC endpoint
+    ///
+    /// - Warning: If more than one blog is found, they'll be considered duplicates and be
+    /// deleted leaving only one of them.
+    ///
+    /// - Parameters:
+    ///   - xmlrpc: the XML-RPC endpoint URL as a string
+    ///   - account: the account the blog belongs to
+    ///   - context: the NSManagedObjectContext containing the account and the Blog object.
+    /// - Returns: the blog if one was found, otherwise it returns nil
+    static func lookup(xmlrpc: String, andRemoveDuplicateBlogsOf account: WPAccount, in context: NSManagedObjectContext) -> Blog? {
+        let predicate = NSPredicate(format: "xmlrpc like %@", xmlrpc)
+        let foundBlogs = account.blogs.filter { predicate.evaluate(with: $0) }
+
+        if foundBlogs.isEmpty {
+            return nil
+        }
+
+        if foundBlogs.count == 1 {
+            return foundBlogs.first
+        }
+
+        // If more than one blog matches, return the first and delete the rest
+
+        // Choose blogs with URL not starting with https to account for a glitch in the API in early 2014
+        let blogToReturn = foundBlogs.first { $0.url?.starts(with: "https://") == false }
+            ?? foundBlogs.randomElement()!
```

Comment:
It's been almost ten years since the glitch mentioned here. I wonder whether it's still necessary to account for it?

Also, the comment says that we're picking the first matching blog, but if all blogs have HTTPS URLs, we'll fallback to a random one. 

Would it be enough to return `first!`, or is there a reason for this?

```suggestion
        // Choose blogs with URL not starting with https to account for a glitch in the API in early 2014
        let blogToReturn = foundBlogs.first { $0.url?.starts(with: "https://") == false }
            ?? foundBlogs.first!
```

Reply:
> I wonder whether it's still necessary to account for it?

I don't know enough to be confident to delete this `url?.starts(with: "https://")` lookup criteria 🥲 

> Would it be enough to return first!, or is there a reason for this?

The original Objective-C code used `anyObject`, which can be translated to `randomElement`. I guess, technically, `first` is same as `randomElement` since `foundBlogs` is a `Set`, but maybe it's also good to explicitly say a random blog is chosen, instead of the first one (whatever that means)?